packit_service/worker/helpers/testing_farm.py

```diff
@@ -60,40 +61,92 @@ class CommentArguments:
-            self.packit_command = match.group("packit_command")
-            logger.debug(f"Parsed packit_command: {self.packit_command}")
+            arguments_str = match.group(1)
+        else:
+            # If the command prefix is not found, nothing to parse
+            logger.debug("Command prefix not found in the comment.")
+            return
 
-        match = re.search(r"(--identifier|--id|-i)[\s=](?P<identifier>\S+)", comment)
-        if match:
-            self.identifier = match.group("identifier")
-            logger.debug(f"Parsed test argument -> identifier: {self.identifier}")
+        # Use shlex to split the arguments string into a list
+        args_list = shlex.split(arguments_str)
+        logger.debug(f"Arguments list after shlex splitting: {args_list}")
 
-        match = re.search(r"--labels[\s=](?P<labels>\S+)", comment)
-        if match:
-            self.labels = match.group("labels").split(",")
-            logger.debug(f"Parsed test argument -> labels: {self.labels}")
+        # Parse known arguments
+        try:
+            args, unknown_args = self.parser.parse_known_args(args_list)
+            logger.debug(f"Parsed known args: {args}")
+            logger.debug(f"Unknown args: {unknown_args}")
+        except argparse.ArgumentError as e:
+            logger.error(f"Argument parsing error: {e}")
+            return
 
-        match = re.search(r"(?P<pr_arg>[^/\s]+/[^#]+#\d+)", comment)
-        if match:
-            self.pr_argument = match.group("pr_arg")
-            logger.debug(f"Parsed test argument -> pr_argument: {self.pr_argument}")
+        self.parse_known_arguments(args)
+        self.parse_unknown_arguments(unknown_args)
+
+    @property
+    def parser(self):
+        if self._parser is None:
+            # Set up argparse
+            self._parser = argparse.ArgumentParser()
+            self._parser.add_argument("packit_command")
+            self._parser.add_argument("--identifier", "--id", "-i")
+            self._parser.add_argument("--labels", type=lambda s: s.split(","))
+            # Allows multiple --env arguments
+            self._parser.add_argument("--env", action="append")
+
+        return self._parser
+
+    def parse_known_arguments(self, args: argparse.Namespace) -> None:
+        # Assign the parsed arguments to the class attributes
+        self.packit_command = args.packit_command
+        logger.debug(f"Parsed packit_command: {self.packit_command}")
+
+        self.identifier = args.identifier
+        logger.debug(f"Parsed identifier: {self.identifier}")
+
+        if args.labels:
+            self.labels = args.labels
+            logger.debug(f"Parsed labels: {self.labels}")
+
+        if args.env:
+            self.envs = {}
+            for env in args.env:
+                if "=" in env:
+                    key, value = env.split("=", 1)
+                    self.envs[key] = value
+                    logger.debug(f"Added env variable: {key}={value}")
```

Comment:
```suggestion
                    logger.debug(f"Parsed env variable: {key}={value}")
```

just for consistency